soc/arm/silabs_exx32/common/soc_power.c

```diff
@@ -24,6 +69,52 @@ __weak void pm_state_set(enum pm_state state, uint8_t substate_id)
 
 	LOG_DBG("SoC entering power state %d", state);
 
+#ifdef CONFIG_SOC_GECKO_DEV_INIT
+
+	uint32_t BasePRI;
+
+	/* WorkAround: When this function is entered the Kernel has disabled
+	 * interrupts using BASEPRI register. This is incorrect as it prevents
+	 * waking up from any interrupt which priority is not 0. Work around the
+	 * issue using __set_BASEPRI(0) to remove the BASEPRI masking
+	 */
+
+	BasePRI = __get_BASEPRI();
+	__set_BASEPRI(0);
```

Comment:
not sure I follow here, please explain this workaround better. It may be better to fix/adjust common code.